contrib_bots/test-bots

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import sys
+import unittest
+
+from unittest.mock import MagicMock, patch
+from unittest import TestCase
+
+from run import get_lib_module
+from bot_lib import BotHandlerApi, StateHandler
+
+class BotTestCase(TestCase):
+
+    def mock_test(self, messages, message_handler):
+        # type: (List[Dict[str, str]], Function) -> None
+        # Mocking BotHandlerApi
+        with patch('__main__.BotHandlerApi') as MockClass:
+            instance = MockClass.return_value
+
+            for message in messages:
+                # Send message to the concerned bot
+                message_handler.handle_message(message, MockClass(), StateHandler())
+
+                # Check if BotHandlerApi is sending a reply message.
+                # This can later be modified to assert the contents of BotHandlerApi.send_message
+                assert instance.send_message.called is True
```

Comment:
We would also want to verify the response from Zulip server. In our case, we should be able to specify return value of `send_message` so that we can check what the code of bot does against different responses from Zulip server. Take a look at `bounce_request` function that we use to mock out push notification servers, https://github.com/zulip/zulip/blob/master/zerver/tests/test_push_notifications.py#L77-L90.